util-core/src/main/scala/com/twitter/util/Batcher.scala

```diff
@@ -0,0 +1,24 @@
+package com.twitter.util
+
+/** Provides a clean, lightweight interface for working with a batched [[com.twitter.util.Future]] */
+class Batcher[In, Out] private[util](
+  executor: BatchExecutor[In, Out]
+)(
+  implicit timer: Timer
+) extends Function1[In, Future[Out]] { batcher =>
+  /**
+   * Enqueues requests for a batched [[com.twitter.util.Future]]
+   *
+   * Processes all enqueued requests after the batch size is reached
+   */
+  def apply(t: In): Future[Out] = executor.enqueue(t)
+
+  /** Immediately processes all unprocessed requests */
+  def flushBatch(): Unit = {
```

Comment:
* I did not implement the BatchExecutor, but I make no assumptions about order of execution of the individual requests within a batch.  Since there is a one-to-one correlation between the requests I submit and the Futures returned, I as a client can at least decide what order I want to process the results in, regardless of what order they complete.

* As far as why I synchronize, the inline documentation for BatchExecutor tells me I must do so when calling .flushBatch() (BatchExecutor.scala line 107).

* Finally, how do we prevent other callers from using the executor incorrectly: by making BatchExecutor private\[util\] (this was done already by whoever implemented BatchExecutor) and only providing access to it through the Batcher interface (which is what I am adding).

If there is some part of what I've just explained here that you would like me to document in the code, please let me know.